lighthouse-core/lib/page-functions.js

```diff
@@ -98,19 +102,21 @@ function checkTimeSinceLastLongTask() {
  * @template {string} T
  * @param {T} selector Optional simple CSS selector to filter nodes on.
  *     Combinators are not supported.
- * @return {Array<HTMLElementByTagName[T]>}
+ * @return {Array<ParseSelector<T>>}
  */
 function getElementsInDocument(selector) {
   const realMatchesFn = window.__ElementMatches || window.Element.prototype.matches;
-  /** @type {Array<HTMLElement>} */
+  /** @type {Array<ParseSelector<T>>} */
   const results = [];
 
-  /** @param {NodeListOf<HTMLElement>} nodes */
+  /** @param {NodeListOf<Element>} nodes */
   const _findAllElements = nodes => {
     for (let i = 0, el; el = nodes[i]; ++i) {
       if (!selector || realMatchesFn.call(el, selector)) {
+        // @ts-expect-error - el is verified as matching above, tsc just can't verify it through the .call().
```

Comment:
In personal projects I've started only ever adding `@ts-expect-error` to lines that are `const thing: Bar = foo` based on the number of times we've been screwed by ignoring the wrong errors :) I'm curious what your stance is on that.

I don't think this case is particularly dangerous, it just made me think of it.

Reply:
No, you're definitely right it's better, and maybe we should establish that as the pattern...the only error being suppressed would be the type change of `el`, while when it's on this line, it could also be `results` has the wrong type or whatever.

My only hesitation is how annoying jsdoc is for this, turning two lines into three :) If only there was a `// @ts-expect-error-line` or something...

```js
if (!selector || realMatchesFn.call(el, selector)) {
  /** @type {ParseSelector<T>} */
  // @ts-expect-error - el is verified as matching above, tsc just can't verify it through the .call().
  const matchedEl = el;
  results.push(matchedEl);
}
```

worth it?

Reply:
> worth it?

(yes)

Reply:
haha, sure. I was mostly asking for future reviews if you're onboard with that policy too, but here is good :)